tests/testthat/test_plotDispersion.R

```diff
@@ -4,39 +4,193 @@ context("DGEobj.plots - tests for plotDispersion.R functions")
+                                designMatrix  = designMatrix,
+                                symbolSize    = 10,
+                                symbolShape   = "triangle",
+                                symbolcolor   = red)
+    expect_s3_class(plot_disp, c("canvasXpress", "htmlwidget"))
+
+    plot_disp <- plotDispersion(DGEdata       = t_obj1$counts,
+                                plotType      = "ggplot",
+                                designMatrix  = designMatrix,
+                                plotCategory  = "BCV",
+                                symbolSize    = 3,
+                                symbolShape   = "square",
+                                symbolColour  = "red" )
+    expect_s3_class(plot_disp, c("gg", "ggplot"))
+
+    #Testing lineFit parameters
+    plot_disp <- plotDispersion(DGEdata      = dgelist,
+                                designMatrix = designMatrix,
+                                lineFit      = "loess",
+                                lineType     = "dotted",
+                                linefitColor = "red")
+    expect_s3_class(plot_disp, c("canvasXpress", "htmlwidget"))
+
+    plot_disp <- plotDispersion(DGEdata      = dgelist,
+                                designMatrix = designMatrix,
+                                lineFit      = "loess",
+                                lineType     = "dotted",
+                                linefitColor = "red",
+                                plotType     = "ggplot")
+    expect_s3_class(plot_disp, c("gg", "ggplot"))
+
+    plot_disp <- plotDispersion(DGEdata      = dgelist,
+                                designMatrix = designMatrix,
+                                lineFit      = "loess",
+                                lineType     = 1,
+                                linefitColor = "red",
+                                plotType     = "ggplot")
+    expect_s3_class(plot_disp, c("gg", "ggplot"))
+
     # testing assert statements
     expect_error(plotDispersion(),
-                 regexp = "DGEdata must be specified.")
+                 regexp = "Both DGEdata and designMatrix must be specified.")
     expect_error(plotDispersion(DGEdata = t_obj1$counts),
-                 regexp = "designMatrix must be specified and should be of class 'matrix'.")
+                 regexp = "Both DGEdata and designMatrix must be specified.")
     expect_error(plotDispersion(DGEdata = dgelist, designMatrix = designMatrix, plotCategory = "cx"),
-                 regexp = "Plot value must be either dispersion or BCV.")
+                 regexp = "Plot Category must  either be dispersion or BCV.")
     expect_error(plotDispersion(DGEdata = dgelist, designMatrix = designMatrix, plotType = "cx"),
-                 regexp = "Plot type must be either canvasXpress or ggplot.")
+                 regexp = "Plot type must either be canvasXpress or ggplot.")
+
+    #testing warning messages
+    expect_warning(plotDispersion(DGEdata      = dgelist,
```

Comment:
you need also to check if the method has returned the plot correctly in addition to the warning message

Reply:
These tests are only to check the warning messages. Since the default values are being assigned they will render the regular plots only. I am wondering if we should check if plot is returned for all the warning messages.

Reply:
I think yes, to make sure that the plot is what is returned and not the message